WordPress/Classes/Utility/DisplayableImageHelper.m

```diff
@@ -0,0 +1,184 @@
+#import "DisplayableImageHelper.h"
+
+static const NSInteger FeaturedImageMinimumWidth = 640;
+
+@implementation DisplayableImageHelper
+
+/**
+ Get the url path of the image to display for a post.
+
+ @param dict A dictionary representing a posts attachments from the REST API.
+ @return The url path for the featured image or nil
+ */
++ (NSString *)searchPostAttachmentsForImageToDisplay:(NSDictionary *)attachmentsDict
+{
+    NSArray *attachments = [[attachmentsDict dictionaryForKey:@"attachments"] allValues];
+    if ([attachments count] == 0) {
+        return nil;
+    }
+
+    NSString *imageToDisplay;
+
+    attachments = [self sanitizeAttachmentsArray:attachments];
+    NSSortDescriptor *descriptor = [[NSSortDescriptor alloc] initWithKey:@"width" ascending:NO];
+    attachments = [attachments sortedArrayUsingDescriptors:@[descriptor]];
+    NSDictionary *attachment = [attachments firstObject];
+    NSString *mimeType = [attachment stringForKey:@"mime_type"];
+    NSInteger width = [[attachment numberForKey:@"width"] integerValue];
+    if ([mimeType rangeOfString:@"image"].location != NSNotFound && width >= FeaturedImageMinimumWidth) {
```

Comment:
What if the first attachment is an HD video and its width is greater than the largest image? I'd filter by mime type first, then sort